lib/lotus/model/adapters/sql/command.rb

```diff
@@ -55,10 +59,41 @@ def update(entity)
           # @api private
           # @since 0.1.0
           def delete
-            @collection.delete
+            _rescue_database_error do
+              @collection.delete
+            end
           end
 
           alias_method :clear, :delete
+
+          private
+
+          # Rescues Sequel::DatabaseError in yielded block
+          #
+          # @api private
+          def _rescue_database_error
+            yield
+          rescue Sequel::DatabaseError => e
+            raise _transform_database_error(e)
```

Comment:
@jodosha would you please explain the stance you are taking on this and the reasoning behind it. I would love to understand your thoughts and reasoning both from the pragmatic side of things as well as the purest design side of things so that I and others can make these decisions prior to PR in the future.